Aarch64 muscle built from source fails

Hello,

Since only a Mac arm64 version is provided, I tried to build my own for a distribution on aarch64 ubuntu.

The build finished with zero errors, but when I ran muscle with no parameters I git this:

```
---Fatal error---
myutils.cpp(122) assert failed: sizeof(void *) == 4
```

By the way at Bioconda Linux ARM64 [CI](https://app.circleci.com/pipelines/github/bioconda/bioconda-recipes/110998/workflows/5dc50569-ee79-4976-80f3-885ae4cf2406/jobs/181817?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link&utm_content=summary) I get this error:
```
12:37:16 [32mBIOCONDA INFO[0m (OUT) $BUILD_PREFIX/bin/aarch64-conda-linux-gnu-c++ -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem $PREFIX/include -DNDEBUG -pthread -fvisibility-inlines-hidden -fmessage-length=0 -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O3 -pipe -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/muscle-5.1 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix -O3 -fopenmp -ffast-math -c -o Linux/testlog.o testlog.cpp[0m
12:37:16 [32mBIOCONDA INFO[0m (OUT) In file included from testlog.cpp:2:[0m
12:37:16 [32mBIOCONDA INFO[0m (OUT) In function 'uint64_t GetClockTicks()',[0m
12:37:16 [32mBIOCONDA INFO[0m (OUT)     inlined from 'void cmd_testlog()' at testlog.cpp:71:26:[0m
12:37:16 [32mBIOCONDA INFO[0m (OUT) timing.h:26:9: error: impossible constraint in 'asm'[0m
12:37:16 [32mBIOCONDA INFO[0m (OUT)    26 |         __asm__ __volatile__ ("rdtsc" : "=a" (lo), "=d" (hi));[0m
12:37:16 [32mBIOCONDA INFO[0m (OUT)       |         ^~~~~~~[0m
12:37:16 [32mBIOCONDA INFO[0m (OUT) In function 'uint64_t GetClockTicks()',[0m
12:37:16 [32mBIOCONDA INFO[0m (OUT)     inlined from 'void cmd_testlog()' at testlog.cpp:83:26:[0m
12:37:16 [32mBIOCONDA INFO[0m (OUT) timing.h:26:9: error: impossible constraint in 'asm'[0m
12:37:16 [32mBIOCONDA INFO[0m (OUT)    26 |         __asm__ __volatile__ ("rdtsc" : "=a" (lo), "=d" (hi));[0m
12:37:16 [32mBIOCONDA INFO[0m (OUT)       |         ^~~~~~~[0m
```

I don't get this error at Github Actions and on my local build ...

--- a/.github/workflows/build_linux.yml
+++ b/.github/workflows/build_linux.yml
@@ -1,6 +1,12 @@
 name: build_linux
 
 on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
   # Enable manual run from the Actions tab
   workflow_dispatch:
 
@@ -27,3 +33,35 @@ jobs:
         with:
           name: muscle-linux-binary
           path: src/Linux/muscle
+
+  build-aarch64:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+
+    - name: Build
+      uses: uraimo/run-on-arch-action@v2
+      with:
+        arch: aarch64
+        distro: ubuntu20.04
+        githubToken: ${{ github.token }}
+        dockerRunArgs: |
+          --volume "${PWD}:/muscle"
+        install: |
+          apt-get update -q -y
+          apt-get install -q -y make gcc g++ file git
+        run: |
+          cd /muscle  
+          uname -a
+          lscpu
+          cd src/
+          make LDFLAGS2=-static
+          file src/Linux/muscle | grep aarch64 
+
+    - name: Upload binary artifact
+      uses: actions/upload-artifact@v2
+      with:
+        name: muscle-linux-aarch64-binary
+        path: src/Linux/muscle
+    

--- a/.github/workflows/build_linux.yml
+++ b/.github/workflows/build_linux.yml
@@ -10,13 +10,22 @@ on:
   # Enable manual run from the Actions tab
   workflow_dispatch:
 
+permissions:
+  contents: read    
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+    
+
 jobs:
   build:
+    name: Linux x86_64
     runs-on: ubuntu-20.04
 
     steps:
       # Checks-out the repository under $GITHUB_WORKSPACE
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
 
       # Runs a set of commands using the runners shell
       - name: Build-commands
@@ -27,14 +36,16 @@ jobs:
           cd $GITHUB_WORKSPACE
           ls -lh
           cd src
-          make LDFLAGS2=-static
+          make -j LDFLAGS2=-static
+          file Linux/muscle | grep x86-64
       - name: Upload binary artifact
-        uses: actions/upload-artifact@v2
+        uses: actions/upload-artifact@v4
         with:
           name: muscle-linux-binary
           path: src/Linux/muscle
 
   build-aarch64:
+    name: Linux aarch64
     runs-on: ubuntu-latest
     steps:
     - name: Checkout
@@ -50,17 +61,18 @@ jobs:
           --volume "${PWD}:/muscle"
         install: |
           apt-get update -q -y
-          apt-get install -q -y make gcc g++ file git
+          apt-get install -q -y make g++ file git
         run: |
+          git config --global --add safe.directory /muscle
           cd /muscle  
           uname -a
           lscpu
           cd src/
-          make LDFLAGS2=-static
-          file src/Linux/muscle | grep aarch64 
+          make -j LDFLAGS2=-static 
+          file Linux/muscle | grep aarch64
 
     - name: Upload binary artifact
-      uses: actions/upload-artifact@v2
+      uses: actions/upload-artifact@v4
       with:
         name: muscle-linux-aarch64-binary
         path: src/Linux/muscle

--- a/src/myutils.h
+++ b/src/myutils.h
@@ -11,7 +11,7 @@
 #define TRACK_ALLOC		0
 #define ALLOC_TOTALS	0
 
-#if defined(__x86_64__) || defined(_M_X64) || defined(__arm64__)
+#if defined(__x86_64__) || defined(_M_X64) || defined(__arm64__) || defined(__aarch64__)
 #define	BITS			64
 #else
 #define	BITS			32

